### ISSUE TITLE
gist of "dividend pathways" (useless in it's current form)

### ISSUE THREAD

**Issue by serapath** [serapath/economy#14]
-------------------------------------------------------------------------------
# RESILIENCE.ME

---
if (dividendPathways[_node][i].amount - _taxColected > 0) {
  dividendPathways[_node][i] -= _taxCollected
}
else removeDividendPathway(_node, i)

function transfer (address _to, uint256 _value) {
  /* if the sender doesnt have enough balance then stop */
  if (balanceOf[msg.sender] < _value) throw
  /* calculate tax */
  uint256 taxCollected = _value * taxRate / 1000
  /* create the dividend pathway */
  dividendPathways[_to].push(dividendPathway({
    from: msg.sender,
    amount: _value,
    timestamp: now
  }))
  someRedistribution(_to, taxCollected)
}
---

PREDICTION:
1. DACs (decentralized autonomous corporations) who choose to pay out highest dividends might inherit the earth

GOAL:
1. To create a welfare system based on the same peer-to-peer principles as Ripple.

RESILIENCE = insurance company + kickstarter + reward mechanisms + UBI
=> A free market, incentive-based, welfare system.

"incentive based voluntary taxation" => freed from hierarchical social structure inherent in coercive force based systems

basicincome = WELFARE

PREREQUISITE:
1. users can't conceal transactions from the system
   * connected accounts of any currency need to be tracked
   * multiple connected accounts under different identities are ok
   * users need to pay in first to get paid out at all

MOTIVATION:
The idea that I want to replicate to you is that p2p-dividend protocols
should be designed so that each cultural cluster feeds back on it´s own emotional intent.
Each swarm - what we used to call nation - supports its own cultural resilience.

This ties in with my swarm-redistribution algorithm and the idea that everyone becomes a shareholder of every investment they make,
 Every single transaction you ever make and have ever made is treated as an investment,
 and creates what I call a dividend pathway, and you receive dividends through these pathways.
 In other words, if you create a pathway of $100, you´ll receive $100s in shares, and then that pathway closes.

------------------
@THOUGHTS:

I INVEST WORK and/or CAPITAL and get nothing for it now
I hope to get DIVIDENDS in the future
I take risk

I BUY GOODS and SERVICES to get something now
I don't get DIVIDENDS in the future
I take no risk and invest nothing

A company prices in some surplus to

ARA ONE
=> buy now, get dividends until depreciated (make depriciation rate public)
==> decaying equity
==> decaying voting right

i can then again give myself a higher payout as entrepreneur (my costs)
i use those to invest into my company instead of my customers

publicly traded companies are that which you can buy into if you wanted

But as a model it would work well, because people buy already
therefore everyone would automatically save for old age
no opting out of that, but people choose where and how and what

---------------------

AARONVAN_W
Ok so this supermarket in my example is "taxed" extra in your system
in order to pay for the basic incomes. But wouldn't that mean that
everything for sale in this supermarket is much more expensive than it
is in supermarkets that are not part of this system? And if so, then
why would anyone shop at this supermarket in the first place?

I guess my question is this: your system sounds a bit like a
donation-based basic income. But then why would we need supermarkets
(and other stores) involved at all?

@AaronvanW
Will you still get basic income if you go to stores that don't support
basic income?
Picture

Johan
No, you will miss out on it, but only temporarily.
The design I have right now is that nodes get disconnected if they consume outside the network,
and stay disconnected until they´ve missed out on shares equal to the amount they consumed for.
In other words, if they consume for $100 from a store outside the network, they miss out on $100 in dividends, and they might loose costumers while they´re disconnected, as their costumers might prefer not to become disconnected.

-----------------

SYSTEM:
0. SUPPLIERS INCENTIVES
   * gain selective advantage
   * shops are more expensive but user build up welfare claims
   * => NODEs are incentivized to join network to gain + keep customers
1. CUSTOMERS INCENTIVES
   * prefer companies who pay basic income
   * NODEs who want to receive WELFARE "should" transact with NODEs who participate in tax system
   * ==> NODEs can create "resilient financial safety" through making active consumerist choices
   * ==> NODEs take greater fnancial responsibility for entire supply line
   * ===> powerful democratic tool/instrument for wealth redistribution
2. USERS have individual "pool"
   * pool size is determined by size of users swarm and swarms redistribution level
     * swarm is all nodes connected to through past (taxed?) transactions
     * swarm includes all nodes directly or indirectly connected to nodes transacted with
     * e.g. if a swarm node is a company, it includes their entire supply chain
     * Basically:
       * users pay for stuff and get all their money back with a delay if the business continues to exist
       * users do not have customers, so nobody ever gets money from them - they are a sink
3. "redistribution level" is determined by tax level set by node and node's swarm
   * node can never receive dividends at higher levels than the one set for their own transactions

* trade with node with lower tax level, distribution level caps to that nodes level until amount of transaction has skipped node
* trade with a node outside the network (tax levle 0%) means to stop receiving dividends for a short while

HOW IT WORKS:
dividend pathways
who gives what to who
api lets you sign dividends
or update yoru dividend rates

-------------

1. EVERY TRANSACTION between A and B in a network OPENS DIVIDEND PATHWAY between them
2. dividend pathway connects A to B and nodes connected to B and their nodes (= This is "meta" and "recusive")
  => this puts value producers in feedback loops with their consumers
3. If you create a pathway to a new node, at say $100, then you´ll receive $100 via that pathway and then it fades.

SCENARIO: Entity A
1. If A buys for $100 from B outside the network, A gets disconnected for duration it takes for $100 dividends to flow through A
2. [LAW OF RESILIENCE] if NODE gets disconnected all their CONSUMERS get disconnected too, and so on...
  --> simple rule to get selective advantage to those who want to create a basic income ecology
  --> paterns of emergent feedback loops between producers and consumers
  --> if you have a successful business and choose to buy from outside the network
  --> loose customers who might want to stay conencted to the network
  DISCONNECTION:
  ---> disconnection is only temporary
  => for duration it takes to receive the TENS equal to the amount you consumed from
  if consume from 1000 from node outside the system
  => disconnects you until everyone else has received 1000 in UBI
3. Staying connected makes a NODE attractive to others who want to stay connected too
4. getting disconnected makes a NODE unattractive for "customers" to stay in touch, to not get disconnected too
  => when disconnected you might also loose consumers who want to stay connected

traffic routing along dividend pathways

MECHANISM:
* choose value producers that provide basic income  =>  receive tax
* pay out dividends to your swarm                   =>  pay tax

------------------------------------------------------------------

1. i get access to your network of users the moment i make a transaction with you
2. networking effect in a transactional way
3. if you make a transaction with a corporation that doesnt pay tax
   * for that amount - until it has cleared - you won't get any income
4. puts it into blocks - takes into account what you have paid and whether you've paid tax on
   * based on that it'll give you something back
5. life in a network sense: i interact with you,  i have effect an you, you have effect on somebody else
   * replicate/ripple through the entire network
6. ppl sustain their societies through their contributions and fees
   * society in turn looks after them
7. CONSUMERS, by seeking out COMPANIES that pay "high dividends" (act as "miners") in the globalmarket,
   * driven by the incentive of BASIC INCOME as a REWARD
8.

--
RULE 1: -> incentivize ventures to purchase from industries who pay taxes
in order for the ventures to be able to pay taxes

if A buys for 1000 from B who uses 0.4% tax rate
=> limites your tax rate for 0.4% for the next 1000$

RULE 2; incentivizes consumers to choose ventures who pay taxes in order for consumer to be able to pay taxes
Dividend received by A is limited by the tax paid by A

if you paid 0.4% tax for previous 1000
=> your dividend flow will be limited to 0.4% for next 1000 (that flows by swarm distribution

RULE 3: disincentivize consumer who pay lowest tax rates - they receive exponentially less basic income
=> up to the point where paying low taxes becomes financially unrewarding

AGGREGATE EFFECT of RULE 1, 2 & 3
1. incentives for consumers to pay tax (consumers prefer ventures that allow them to pay tax)
  * => natural selection: companies who want to pay tax have to choose industries who allow them to pay tax
2.

If you want to be part of the system, then you need to let the system read your transactions.
Every transaction you receive and every transaction you send.

PREREQUISITE:
1. need to be able to read accounts transactions
2.

If I've created multiple dividend pathways to one node, but in different dividendRates ?

If a node has multiple dividend pathways, it will use the most efficient one.

See

filter_dividend_pathways_by_dividendRate()
in swarm_redistribution.js

If some companies exist which would pay tax if it was voluntary, this system can work
1. those companies will be attractive to consumers

2. users can choose to donate part of their dividends to differenct causes
=> if companies support their users, those users might support less fortunates too
  * e.g. UBI, school, healthcare

if IOUs represent trust, then coercive taxation is taking trust from somebody

“I trust you to pay me back, but if you can’t pay me back, then I’ll gradually let this debt decay”?
To say that “I’ll help you now, and if you face scarce times with little growth, then i’ll free you of your debt. “

Eureka

I think i've solved the optimization layer.

Every time a transaction is registred, the system registers how much the node should give as dividend,
and also how much each node in that node's swarm should recieve from that dividend.
Each node then keeps track of how much dividends they've recieved.
This database-structure makes it simple.

Object 1
- ...
- ...
- ...
Object 2
- ID
- type: accumulated_dividend
- currency: EUR
- amount: 22
Object 3
- ID
- account: 12345
- curency: EUR
- taxRate: 0.02
- type: dividend_pathway
- total_pathway: 140

APPLICATION LOGS
{
  ...
  type: 'accumulated_dividends',
  accumulated_amount: 0.05
}
{ ID,
  currency: 'RES',
  type: 'accumulated_dividends',
  accumulated_amount: 0.05
}

Then, once that's implemented, and you receive a payment and give a dividend on it,
the full dividend goes to the node who's accumulated the most dividends.
And then someone else in the swarm pays a dividend, the system scans his swarm
and finds the one with most accumulated dividends, and the full dividend goes to it.

This is played out over and over again.
Each node sends just one dividends transactions,
but every node still pays and receives as much as if every node sent to every node.
It's a brilliant solution.

Step by step:

You consume, and send an IOU to the other node. That other node makes a small amount of profit, and shares it as dividends.

That other node then uses that IOU to consume from another node who accepts your IOU. That node makes some profit and shares it as dividends.

That other node then uses that IOU to consume from another node who accepts your original IOU, and makes profit that is shared as dividends.

Your IOU eventually circles back to you, and the loop closes. An entire circle of production and consumption is finished. A ripple. You consumed the output of another node, and promised that other node something in return. It´s first once the ripple-cycle is finished that you´re back at equilibrium. The profit that´s been collected in each step has been shared with everyone in the IOU network, and every node has felt safe throughout the entire cycle of production and consumption.

DIVIDEND PATHWAYS
Each transaction A makes creates/adds to one pathway.
Each pathway facilitates a flow of dividends up to the amount that was transacted to create the pathway
If you send me $100, up to $100 in dividends will flow through that pathway.

peer to peer financial security

The dividends are divided on everyone that's connected through these pathways.
I call this a swarm, and I've called the algorithm itself swarm-redistribution or p2p-dividend protocols.
The flow of dividends is then optimized through an optimization layer,

forum.ethereum.org
https://forum.ethereum.org/discussion/comment/7759/#Comment_7759

You might be interested in my system. I designed it 2012, and started developing 2014, and Bitnation have acquired it and it's received some media attention.

Picture

It uses a concept called dividend pathways. Each transaction you make creates or adds to one of these pathways. Each pathway facilitates a flow of dividends up to the amount that was transacted to create the pathway. If you send me $100, up to $100 in dividends will flow through that pathway.

The dividends are divided on everyone that's connected through these pathways. I call this a swarm, and I've called the algorithm itself swarm-redistribution or p2p-dividend protocols.

The flow of dividends is then optimized through an optimization layer,

You can see this optimization on the GIF below in real time,

It uses an inventive structure, a sort of reward system,
or you could think of it as a governance system
that emerges from the feedback loops between producers and consumers,

CO-OP uses this meta pattern for a long time already:
* This meta-pattern, the incentive layer, is similar to what the co-operative business movement has used for decades,

"The dividend scheme makes it more advantageous to be a loyal Coop member.
As more people realize the economic benefits, the idea is that more people will become members,
more people will shop at Coop and that casual customers will transform
into regulars and consolidate their purchases to Coop.
This will make Coop more profitable, making it possible to further improve the stores and offers to members. "

“This means that consumers who want to receive basicincome
will benefit greatly from seeking out corporations that are connected to basicincome.co.
Therefore, corporations are incited to join the network
in their effort to gain and keep costumers.
Not only will this mean that consumers can create a resilient financial safety net
through making active consumerist choices. The corporations will also be encouraged
to take greater financial responsibility for their entire supply line.”

---
EXAMPLE: http://www.resilience.me/a-sample-scenario.html

Johan:
if person A consumes from person B for $100, and creates a dividend pathway of $100,
then that pathway conveys $100 in dividends. After $20 dividends has passed through it,
the pathway's shrunk to $80.

A-1000: 1000*0.02=20->B

STEP 1
A pays 1000 to B
B receives 980
and 0.2 tax flow to previous customers of B
=> each person in the "dividend pathways" gets equal amounts of that

STEP 2-X
repeat step 1 as often as needed to pay to "dividend pathways" the 20 until used up.

buying/paying builds up voltage of potential tax flow
=> each account creates a huge build up of distribution voltage
==> through it's unique financial connections
==> it is the voltage that powers it's welfare

BUT RATE LIMIT:
1. once a user has 0 voltage left -> distribution bypasses that user

e.g. users have a CAP, which once reached, will skip them in terms of distrbution

The dividends are based on the dividendRates of:
the dividend pathways,
and the dividend issuer.
If person B has a dividendRate of 0.03, then $3 is payed out as dividends.

If 4 other people, besides person A, has dividend pathways to person B, person C, D, E and F,
and two people have dividend pathways to person A, person G and H,

then the $3 is divided based on the dividendRate of person As, Cs,Ds, Es and Fs dividend pathways,
and person G and H's dividends are a ratio of their pathway and person As pathway.

If C,D,E,and F all have dividend pathways of 0.03,
and person A of 0.02, and person H 0.01, and person G 0.02,
then the dividendRate_quota of each person is:

C:1,D:1,E:1,F:1, A:2/3, G:2/3, H:2/6

Results:

person H gets 5.9% of $3,  $0.177,
person G and A get 5.9%x2, 11.8%, 0.177*2, $0.354
C, D, E, F get 17.6%, $0.531

0.177+0.354*2+0.528*4 = $3

VIDEO_1: https://www.youtube.com/watch?v=cJXS0SoL0iA

Here's part of the swarm-redistribution algorithm.
You pass the transaction tax into compute_swarm(),
it searches through the dividend pathway web (similar to how ripple finds trust-pathways),
and finds all nodes that are inter-connected with the node who paid transaction tax.
The tax is then divided on all these nodes (could be 100s, could be billions),
and nodes who have dividend pathways with higher dividendRates
meaning they've been created when a nodeA consumed from a nodeB who chose a high transaction tax
nodes who have high-yield dividend pathways get a bigger piece of the taxed amount.

The taxed sum is then divided on the 100s or billions of nodes,
each node keeps a log about how much dividends they have received,
and the whole taxed sum is then sent to the node who has accumulated the most in that log.

The node who paid the transaction tax sends the whole amount to one single node,
but all nodes still receive as much as if the money had been divided on 100s or billions of nodes.

The protocol P2PDP (P2P-dividend protocol) is a set of rules for adding dividends to a financial transaction.
The invention of P2PDP opens up a world of possibilites.

VIDEO_2: https://www.youtube.com/watch?v=Yo4Fs3nnA0U

FIRST LAW: incentivize ventures to purchase from industries who pay taxes in order for the ventures to be able to pay taxes
1. nodeA buys for 1000 from nodeB(0.4%taxrate)
   => limits nodeA taxrate to 0.4% for next received 1000
   => affects amount of basic income you receive

2. dividend voltage is limited by tax you pay
   => 0.4% * 1000 spent limits to receive 1000 at 0.4% from swarm distribution

SECOND LAW:
incentivizes consumers to choose ventures who pay taxes in order for consumers to be able to pay taxes

THIRD LAW:
creates sharp diincentive for consumers who pay lowest taxes
they receive exponentially less basic income up to the point where paying low taxes becomes financially unrewarding

Aggregate effect: laws feedback upon each other

--

//collect a liveable basic income from the insurance pool

RESILIENCE consists of 3 layers

1. tax collecting layer -> connects financial network to resilience network

  1. nodeA specified all currencies with tax rates to connect
  2. send date for all incoming + outgoing transactions for those currencies
  //declare taxes for incoming transactions
  {
  "command" : "declare_tax"
  "amount" : 400,
  "currency" : "PFJ",
  "taxRate" : 1010000000,
  }
  //the server needs access to all your transactions
  //if you use ripple, the serveran subscribe via their API
  {
    "command": "subscribe",
    "accounts": [your_account],
    "streams": ["transactions"]
  }
  3. resilience server generates list of outgoing (unsigned) swarm distribution payments
    //sign the list of outgoing payments, IF <= (amount * taxRate)
    {
      "command": "submit",
      "tx_blob": "12000322000000002400000038684000xxxxx"
      "total_amount" : 16
    }

nodeA pays nodeB 1000 at 1.4% tax rate
=> $7

tax collecting layer is integrated with a nodes financial applications

2. swarm distribution layer
shuttles welfare around

tax nodeA pays is sent directly to the swarm of accounts that are connected to nodeA
industry pays out to the swarm of consumers the industry provides for

you receive swarm distributions via all outgoing transactions that were taxes and connected to the network

resilience network geenrates list of payments
you sign them
and tax becomes dividends

3. incentive layer
makes it possible to run decentralized welfare

--
STEPS
1. subscribe to an accounts transaction
2. transaction tax
3. share tax with swarm

INCENTIVES

First law of resilience

There is one limit to an accounts taxRate. The taxRate that is payed for it´s outgoing transactions, id est not the tax the account itself pays, but the tax that accounts that it sends payments to pays, sets an upper limit to taxRate.

If an account makes a payment of 1000 USD and the recipient account pays 0 % tax, then the taxRate for that currency will be limited to <0 % for the next 1000 USD it receives. It will have locked itself out from the network.

If I send you 1000 USD, and you don´t tax anything, then I won´t get a dividend-pathway via that transaction. I won´t get the welfare benefits that the network provides.

This means that ventures that pay taxes will be popular amongst customers. And that industries that pay taxes will be popular amongst the ventures, because it will allow them to pay taxes.

This creates a bottoms-up, recursive, positive feedback loop.

Second law of resilience

There is one limit to an accounts dividend-voltage.

The tax an account pays for incoming transactions sets an upper limit to it´s dividend-voltage.

If I send you 1000 USD, and you don´t tax anything, then the server turns off your dividend for the next 1000 USD that flow to you. after 1000 USD, it turns on that currency again.

If I send you 1000 USD and you tax 1 %, then the server limits your dividend-pathways for that currency to 1 % for the next 1000 USD that flow to you.

This script only applies if you have incoming transactions. If you don´t have any incoming transactions at all, it doesn´t apply. And if your outgoing transasctions exceed your incoming transactions, then it doesn´t apply for the surplus.

it´s possible to have an account, receive 0 incoming transactions, and collect dividends for outgoing transaction. it´s possible to "live on basic income". and that´s the case with any basic income system. that´s the idea of unconditional basic income.

the account would still have to be validated. a person couldn´t keep 1000 accounts and collect basic incomes. the p2p-identification system prevents that. so the same rules apply as with a centralized tax-governance system. people don´t need income to receive unconditional basic income. that´s the idea. they just need to be trusted citizens. basic income is one of their citizen rights. and in this system, citizenship is enforced through p2p-identification.

Third law of resilience

The basic income is set so that the accounts who pay the lowest taxes won´t get a full basic income.

Each account receives dividends up to a time-period-specific limit. The limit is:
the sum of all tax that goes into the peer-network
/ a % of the number of people in the network

ex.

sum of currency a
/ 90 % of the people in the peer-network

sum of currency b
/ 90 % of the people in the peer-network

sum of currency c
/  90 % of the people in the peer-network

...

This means that accounts with high-voltage dividend-connections fill up their basic income first, and the accounts that has the lowest voltage in the peer-network fills up their basic income from what remains.

basic income = (currency a dividend + currency b dividend + currency c dividend...)

The percentage is calculated in some smart way. I don´t really feel that it´s important to write out the details right now. it´s not a hard problem.

Aggregate effect

If you buy a car from me for 10,000 USD, and I tax 0%, then your taxRate will be limited to 0 % for the next 10,000 USD you receive.

If you pay 0% of tax for 10,000 USD, then all your dividend-pathways will turn off for the next 10,000 USD that flow through you. That´s about a year of basic income. Your dividend-flow would be turned off for about a year.

Accounts will want to use >0% taxRate.

//the third law of resilience
//The basic income is set so that the accounts who pay the lowest taxes won´t get a full basic income.
//It´s designed to dis-incentivize paying much-below-average-tax.

Accounts will want to use % of taxRate that give them full basic income.

This means that accounts will want to consume from ventures that pay taxes, because it in turn allows them to pay taxes. And the venture will want to consume from industries that pay taxes, and so on.

If a venture buys cars from an industrial ecosystem that pays really low tax, like 0,1%, then the venture will be limited to paying 0,1 %.

If you buy a car for 10,000 USD from that venture, and it taxes 0,1 %, then you´ll be limited to taxing the next 10,000 USD you receive at 0,1 %. If you do that, and you don´t have a choice if you´ve already bought the car, then your dividend-flow will be limited to 0,1 % for the next 10,000 USD. And since the basic income is set so that low-tax-payers receive less, you´ll loose out on a lot of money.

You won´t want that, so you´ll choose to consume from ventures that allow you to in turn pay your tax, and they will in turn choose to consume from industry ecosystems that allow them to pay.

Think of the dividend pathways and the incentive layer as an alternative method for collecting tax,
a decentralized method for collecting tax. Now, how much of those dividend could you tap into before you start to loose users ?
